src/models/AnnotationEditPage.ts

```diff
@@ -0,0 +1,23 @@
+import { Expect } from '@playwright/test';
+import { DataSourcePicker } from './DataSourcePicker';
+
+import { PluginTestCtx } from '../types';
+import { GrafanaPage } from './GrafanaPage';
+
+export type VariableType = 'Query' | 'Constant' | 'Custom';
+
+export class AnnotationEditPage extends GrafanaPage {
+  datasource: DataSourcePicker;
+  constructor(ctx: PluginTestCtx, expect: Expect<any>) {
+    super(ctx, expect);
+    this.datasource = new DataSourcePicker(ctx, expect);
+  }
+
+  async runQuery() {
+    await this.ctx.page.getByRole('button', { name: 'TEST' }).click();
+  }
+
+  async expectRunQueryResultToContainText(text: string) {
```

Comment:
A note here:
Rather than exposing this expect method in the POM, I would like to create custom expect matcher for this. For example: 
```typescript
expect(<AnnotationEditPage>).toHaveQueryResult(<text>)
```
However, seems it's currently [not possible](https://github.com/microsoft/playwright/issues/22031) to provide type-safe matchers, so probably better to have in the POM for now.